jt-classbreaks/src/main/java/it/geosolutions/jaiext/classbreaks/NaturalBreaksHistogramOpImage.java

```diff
@@ -166,5 +167,21 @@ protected void postCalculate(Classification c, int band) {
         }
         breaks[0] = buckets.get(0).getMin();
         hc.setBreaks(band, breaks);
+        if (percentages.booleanValue()) {
+            ClassPercentagesManager percentagesManager = new ClassPercentagesManager();
+        }
+            hc.setPercentages(
+                    getPercentages(buckets, m, k>breaks.length?breaks.length:k, breaks));
+    }
+
+    private double[] getPercentages(List<Bucket> buckets, int m, int k, Double[] breaks) {
```

Comment:
I don't understand how this is working... each bucket represents a range and the count of values in that range, no?
If you don't use the counts from the buckets, how are percentages computed?

Reply:
fixed